### ISSUE TITLE
ENH: START_SELECTION_WEIGHTS are undocumented and untested for generalisation — add provenance and cross-validation

### ISSUE THREAD

**Issue by roeimed0** [roeimed0/bacterial-gene-prediction#103]
## Problem

\`src/config.py\` defines:

\`\`\`python
START_SELECTION_WEIGHTS = {
    "codon": 4.8562,
    "imm": 1.0107,
    "rbs": 0.6383,
    "length": 7.4367,
    "start": 0.2755,
}
\`\`\`

These weights have 4-decimal precision, suggesting they were optimised on some dataset, but:
1. There is no comment or document explaining their origin
2. It is unknown whether the TEST_GENOMES were part of the optimisation set (data leakage risk)
3. No cross-validation result is documented
4. The asymmetry is large: \`length\` (7.44) and \`codon\` (4.86) dominate; \`start\` (0.28) is near-zero

## Impact

If these weights were optimised on the same genomes used for evaluation, reported metrics are overoptimistic. Even without leakage, a 5-way weight with 4 decimal precision is almost certainly overfitted to the training set.

## Fix

1. Document origin of weights (training set, method, date) in \`config.py\`
2. Perform leave-one-genome-out cross-validation on \`TEST_GENOMES\` to confirm they generalise
3. Compare against uniform weights (all 1.0) — if the gap is small, uniform weights are preferable (simpler, less overfitting risk)
4. Report cross-validation F1 with both weight sets

**Comment by roeimed0** [comment]
Investigated and resolved. Calibration script (scripts/calibrate_start_weights.py) ran coordinate sensitivity analysis on all 100 GENOME_CATALOG genomes. Result: current weights are already at local optimum — all perturbations equal or worse. CE-loss optimizer confirmed regression via bootstrap CI. Calibration report saved to models/calibration_report.json.
